apis/Google.Cloud.Datastore.V1/Google.Cloud.Datastore.V1/DatastoreClientBuilder.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax.Grpc;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Google.Cloud.Datastore.V1
+{
+    // Note: this file would be generated.
+
+    /// <summary>
+    /// Builder class for <see cref="DatastoreClient"/> to provide simple configuration of credentials, endpoint etc.
+    /// </summary>
+    public sealed partial class DatastoreClientBuilder : ClientBuilder<DatastoreClientBuilder, DatastoreClient>
+    {
+        private DatastoreSettings _settings;
+
+        /// <summary>
+        /// Creates a new builder with default options.
+        /// </summary>
+        public DatastoreClientBuilder() : base()
+        {
+        }
+
+        /// <summary>
+        /// Returns a clone of the settings in this builder, or null if no settings have been set. Changes to the
+        /// returned object do not affect the settings in this builder.
+        /// </summary>
+        /// <returns>A clone of the settings in this builder, or null.</returns>
+        public DatastoreSettings CloneSettings() => _settings?.Clone();
```

Comment:
Hmm, I'm uncertain about this.
To match the base class pattern, this should be a property called `Settings`.
I'm guessing that you've done it this way to (a) make it clear by the name that it's a clone of the settings; and (b) made it a method because it's not returning the same instance each time.
However, in some ways I'd prefer it to be the property, then in the xmldoc make it clear that it clones. I found the name `CloneSettings` initially confusing.

Alternatively we could code-gen a second, immutable, settings class; which is returned here. This second class could either use composition, or could be a base class of the mutable settings class...

Reply:
I think I'd like to stick with a method. Would `GetClonedSettings()` be better?